### PR TITLE
[Merged by Bors] - Add an outline on focused elements for accessibility

### LINKED ISSUE #395: Add visual indicator for links on `:focus` event
Making a website usable with keyboard navigation is one of the corner stones of accessibility.

To enable this goal, it's important to know where you currently are while navigating through the website. The Bevy website currently provides no such indications (e.g. on links and the cards), making keyboard navigation almost impossible.

The `:focus` CSS pseudo-class indicates that the element is focused (e.g. by the keyboard).
This is usually indicated by a colored outline around the object.

[Relevant MDN Docs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Keyboard#focusable_element_must_have_focus_styling)

How to reproduce:

- Make sure the website is focused
- Use tab and shift+tab to navigate between the intractable elements. You might see the link preview of your browser changing, but can't visually distinguish the element that's currently focused.

(Tested on Firefox and Chromium)

### PR DESCRIPTION
**Pull request by TimJentzsch**
Fixes #395.

For users that depend on keyboard navigation, it is crucial to have a visual indication of which element is currently focused.

This PR adds a bright blue outline to the focused element:

- Header links:
    
    Focus outline on the header links

- Header buttons:

    Focus outline on the header buttons

- Cards:

    image

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
Can we round the outline on the header links a bit? Those corners are pretty rough.

**Comment by TimJentzsch** [comment]
> Can we round the outline on the header links a bit? Those corners are pretty rough.

Every website that I've ever seen has the hard edges for the focus outlines, but it's probably just because the amount of people that will see them is relatively low.

It looks like this now:
Rounded edges on focus outlines

**Comment by alice-i-cecile** [comment]
I like that!

**Review by alice-i-cecile — APPROVED** [review]

File: sass/_utils.scss [hunk 1]
```diff
@@ -2,8 +2,10 @@
     display: none;
 }
 
-a:focus {
-    outline: none;
+// Visual outline on focused elements, for accessibility
+*:focus {
+    border-radius: 5px;
+    outline: $focus-outline;
```
Comment (plof27) [inline sass/_utils.scss:8]: This is all quite good, but using the `:focus` pseudo-class results in the focus styling being applied whenever the mouse is held down, in addition to when the keyboard focuses an element. This is likely unintentional, and looks a bit odd on most elements (although I kinda like it on the cards).

I'd recommend using `:focus-visible` instead, which is applied only when the keyboard focuses an element (but ignores focus events from other devices, mostly). Further reading [here](https://css-tricks.com/the-focus-visible-trick/) and [here](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-visible).

**Review by plof27 — CHANGES_REQUESTED** [review]
One small styling nitpick, but otherwise this looks good!

File: sass/_utils.scss (hunk 1 above)
Comment (TimJentzsch) [reply sass/_utils.scss:8]: Neat, I didn't know about this!

**Review by plof27 — APPROVED** [review]
Looks great!

**Review by IceSentry — APPROVED** [review]
For some reason I can't change the focus locally with my keyboard., but the code looks good.

**Comment by Nilirad** [comment]
If it had a little bit of transparency wouldn't have been bad IMO.

**Pull request closed without merging**